### PR TITLE
Update "Installation" docs page

### PR DESCRIPTION
**Pull request by shoyer**
- List all optional dependencies
- Clarify when conda-forge is recommended

CC @ocefpaf @jhamman

### CODE REVIEW

**Comment by jhamman** [comment]
lgtm

File: doc/installing.rst [hunk 1]
```diff
@@ -22,6 +22,8 @@ For netCDF and IO
 - `pydap <http://www.pydap.org/>`__: used as a fallback for accessing OPeNDAP
 - `h5netcdf <https://github.com/shoyer/h5netcdf>`__: an alternative library for
   reading and writing netCDF4 files that does not use the netCDF-C libraries
+- `pynio <https://www.pyngl.ucar.edu/Nio.shtml>`__`: for reading GRIB and other
```
Comment (fmaussion) [inline doc/installing.rst]: This link is not working (there is a ` after __)

File: doc/installing.rst [hunk 2]
```diff
@@ -39,31 +41,38 @@ For parallel computing
...
 .. _conda: http://conda.io/
 
-.. note::
-    You might consider using the
-    `conda-forge <https://conda-forge.github.io/>`__ channel, as it has been
-    shown to be more up-to-date and to better handle package dependencies.
+We recommend using the community maintained
+`conda-forge <https://conda-forge.github.io/>`__ channel if you need difficult
+to build dependencies such as cartopy or pynio:
+
```
Comment (fmaussion) [inline doc/installing.rst]: Remove "difficult"

**Comment by fmaussion** [comment]
A bit late but I found too small mistakes

**Pull request merged**